manticore/ethereum/abi.py

```diff
@@ -137,6 +137,8 @@ def _serialize_bytes(value):
     def _serialize_tuple(types, value, dyn_offset=None):
         result = bytearray()
         dyn_result = bytearray()
+        if len(types) != len(value):
+            raise ValueError(f"The number of values to serialize is {'less' if len(value) < len(types) else 'greater'} than the number of types")
```

Comment:
pls use EthereumError for this, we're trying to shift over to that

Reply:
The rest of the code in that file currently uses `ValueError` and I thought that keeping the code locally consistent might be preferable. 

In general, I think it could be quite helpful to use separate exception hierarchies for (Python) programming errors and errors that arise during the normal EVM execution (runtime errors due to e.g. IO errors would be third category). Since `EthereumError` is a subclass of `ManticoreError`, using it for all exceptions in the Ethereum code base increases the risk that programming errors are inadvertently masked. 

For example, currently the Executor logs and suppresses all exceptions, even asserts, which can make it more difficult to diagnose errors and test fixes. Using separate exception hierarchies could make it easier to let exceptions caused by programming errors bubble up to the parent Manticore process/instance.